crates/rspack_core/src/diagnostics.rs

```diff
@@ -258,3 +243,36 @@ pub fn map_box_diagnostics_to_module_parse_diagnostics(
     })
     .collect()
 }
+
+/////////////// Minify error
+
+#[derive(Debug)]
+pub struct MinifyError(pub Error);
+
+impl std::error::Error for MinifyError {
+  fn source(&self) -> ::core::option::Option<&(dyn std::error::Error + 'static)> {
+    Some(<Error as AsRef<dyn std::error::Error>>::as_ref(&self.0))
+  }
+}
+
+impl std::fmt::Display for MinifyError {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    write!(f, "Chunk minification failed:")
+  }
+}
+
+impl miette::Diagnostic for MinifyError {
+  fn code<'a>(&'a self) -> Option<Box<dyn std::fmt::Display + 'a>> {
+    match self.severity().unwrap_or(miette::Severity::Error) {
+      miette::Severity::Advice => unreachable!("miette::Severity::Advice should not be used"),
+      miette::Severity::Warning => Some(Box::new("ChunkMinificationWarning")),
+      miette::Severity::Error => Some(Box::new("ChunkMinificationError")),
```

Comment:
I think all erorr_code should be maintained in a separate file like error_code.json, so it can be easily reviewed otherwise developers may not be aware of this is a public api

Reply:
I would expose some constants. But for generating diagnostics, it's a whole new complex problem that couldn't be done without some hassle. https://github.com/microsoft/TypeScript/blob/069de743dbd17b47cc2fc58e1d16da5410911284/Herebyfile.mjs#L86

Reply:
maybe we don't need generating diagnostics for now, the errorType is limited now so manuallly maintain is acceptable